src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```
@@ -0,0 +1,52 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    @Override
+    public List<FruitTransaction> convertStringsToTransactionsList(List<String> lines) {
+        if (lines == null) {
+            throw new NullPointerException("Input parameter cannot be null");
+        }
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (int i = 1; i < lines.size(); i++) {
+            String currentLine = lines.get(i).trim();
+            if (currentLine.isEmpty()) {
+                continue;
+            }
+            fruitTransactions.add(parseLineToTransaction(currentLine));
+        }
+        return fruitTransactions;
+    }
+
+    private FruitTransaction parseLineToTransaction(String line) {
+        String[] parts = line.split(",");
+        if (parts.length != 3) {
+            throw new RuntimeException("Input line is invalid: has more than 3 parts");
+        }
+        String operationCode = parts[0].trim();
+        String fruitName = parts[1].trim();
+        int quantity;
+        try {
+            quantity = Integer.parseInt(parts[2].trim());
+        } catch (NumberFormatException e) {
+            throw new RuntimeException("Input line is invalid: quantity part is invalid");
```

When parsing the quantity you catch NumberFormatException and rethrow a RuntimeException without including the cause or the offending line. The checklist requires: "Do not ignore exceptions; rethrow as RuntimeException with informative messages and include the cause [CHECKLIST ITEM #8]" and also that "Exception messages should be informative, including context like file path or offending line." Wrap the caught exception and include the `line` string in the message, e.g., `throw new RuntimeException("Invalid quantity in line: '" + line + "'", e);`